controllers/api.js

```diff
@@ -188,7 +178,14 @@ router.post('/payinvoice', async function(req, res) {
           pay_req: req.body.invoice,
         });
 
-        await UserPayee.setPaymentHashPaid(info.payment_hash);
+        // now, receiver add balance
+        let userid_payee = await u.getUseridByPaymentHash(info.payment_hash);
+        if (userid_payee) {
+          let UserPayee = new User(redis, bitcoinclient, lightning);
+          UserPayee._userid = userid_payee; // hacky, fixme
+          await UserPayee.clearBalanceCache();
+          await UserPayee.setPaymentHashPaid(info.payment_hash);
```

Comment:
I think `setPaymentHashPaid` should be outside this IF block, so that payer cant pay same invoice twice if payee cant be found

Reply:
agree, but then a reference payer - payment_hash needs to be created